packages/app-builder-lib/src/configuration.ts

```diff
@@ -32,6 +32,11 @@ export interface Configuration extends PlatformSpecificBuildOptions {
    */
   readonly productName?: string | null
 
+  /**
+   * The executable file name. Defaults to `productName`.
+   */
+  readonly productFilename?: string | null
```

Comment:
This seems to be duplicating what already exists for "Any Linux Target" -> `executableName`
https://www.electron.build/configuration/linux

https://github.com/electron-userland/electron-builder/blob/master/packages/app-builder-lib/src/options/linuxOptions.ts#L24-L28
I think the proper solution would be to move `executableName` out of `LinuxConfiguration` and into `PlatformSpecificBuildOptions`. That would keep it backward compatible for Linux options, while exposing it to each platform to uniquely define.

Reply:
I'm sorry, I don't understand why I'm using executableName on Linux.
(Is this not good with AppInfo.productFilename?)

> I think the proper solution would be to move executableName out of LinuxConfiguration and into PlatformSpecificBuildOptions.

LinuxPackager.executableName will have the same value as AppInfo.productFilename.

Are you ok with this?

Reply:
Your `productFilename` is a nomenclature misnomer as your documentation for it states it is the "executable file name", which is the exact same as `executableName` in `LinuxConfiguration`.

`LinuxConfiguration` inherits from `PlatformSpecificBuildOptions` which is where you're adding this new property to. Please do not do this, it duplicates existing functionality and is confusing as to what value to choose for Linux targets (which takes precedence for instance for Linux builds?)

You should move `executableName` to `PlatformSpecificBuildOptions` (plus update the doc comments on it) and it'll serve your purposes just fine. 
Another way to state this is: 
- Rename this variable to `executableName`
- Remove `executableName` from https://github.com/electron-userland/electron-builder/blob/master/packages/app-builder-lib/src/options/linuxOptions.ts#L24-L28

Reply:
Thanks for the detailed explanation. I'll give it a try.

https://github.com/electron-userland/electron-builder/blob/master/packages/app-builder-lib/src/targets/LinuxTargetHelper.ts#L100

This will affect the value of the exe when it is called FpmTarget.build -> LinuxTargetHelper.writeDesktopEntry -> computeDesktopEntry.

Reply:
> The executable name. Defaults to `productName`.

https://github.com/electron-userland/electron-builder/blob/master/packages/app-builder-lib/src/linuxPackager.ts#L21

I'll going to leave it as it is, but I feel like the default for `LinuxPackager.executableName` is not `productName`. (I think it's only referring to metadata.name)

Reply:
https://github.com/electron-userland/electron-builder/blob/master/packages/app-builder-lib/src/util/macroExpander.ts#L20

productFilename affects not only the executable name, but also the `File Macro`.

The File Macro will keep its original value.

Reply:
> `shortcutName` String - The name that will be used for all shortcuts. Defaults to the application name.

https://github.com/electron-userland/electron-builder/blob/master/packages/app-builder-lib/src/options/CommonWindowsInstallerConfiguration.ts#L76

shortcutName should also use productName. (not filename)

The rest of the code looks fine.